### ISSUE TITLE
Units are added on the wrong position

### ISSUE THREAD

**Issue by douglascdev** [douglas-cpp/snake#26]
After collision is detected with a food, a new unit is added to the snake. The position for it is based on the head's position plus one movement, to where the snake is headed. Before this, though, a step was already done, resulting in two movements. Fix the placement of the new unit to move the snake only once, while still adding another unit.
